Celeste.Mod.mm/Patches/Monocle/Tracker.cs

```diff
@@ -123,5 +123,109 @@ private static List<Type> GetSubclasses(Type type) {
+                        TrackedEntityTypes.Add(type, value);
+                    }
+                    value.AddRange(TrackedEntityTypes.TryGetValue(trackedAs != null && trackedAs.IsAssignableFrom(type)
+                        ? trackedAs : type, out List<Type> list) ? list : new List<Type>());
+                    TrackedEntityTypes[type] = value.Distinct().ToList();
+                }
+                foreach (Type subtype in subtypes) {
+                    if (!subtype.IsAbstract) {
+                        if (!TrackedEntityTypes.TryGetValue(subtype, out List<Type> value)) {
+                            value = new List<Type>();
+                            TrackedEntityTypes.Add(subtype, value);
+                        }
+                        value.AddRange(TrackedEntityTypes.TryGetValue(trackedAs != null && trackedAs.IsAssignableFrom(type)
+                            ? trackedAs : type, out List<Type> list) ? list : new List<Type>());
+                        TrackedEntityTypes[subtype] = value.Distinct().ToList();
+                    }
+                }
+            }
+            else if (typeof(Component).IsAssignableFrom(type)) {
+                StoredComponentTypes.Add(type);
+                if (!type.IsAbstract) {
+                    if (!TrackedComponentTypes.TryGetValue(type, out List<Type> value)) {
+                        value = new List<Type>();
+                        TrackedComponentTypes.Add(type, value);
+                    }
+                    value.AddRange(TrackedComponentTypes.TryGetValue(trackedAs != null && trackedAs.IsAssignableFrom(type)
+                        ? trackedAs : type, out List<Type> list) ? list : new List<Type>());
+                    TrackedComponentTypes[type] = value.Distinct().ToList();
+                }
+                foreach(Type subtype in subtypes) {
+                    if (!subtype.IsAbstract) {
+                        if (!TrackedComponentTypes.TryGetValue(subtype, out List<Type> value)) {
+                            value = new List<Type>();
+                            TrackedComponentTypes.Add(subtype, value);
+                        }
+                        value.AddRange(TrackedComponentTypes.TryGetValue(trackedAs != null && trackedAs.IsAssignableFrom(type)
+                            ? trackedAs : type, out List<Type> list) ? list : new List<Type>());
+                        TrackedComponentTypes[subtype] = value.Distinct().ToList();
+                    }
+                }
+            }
+            else {
+                throw new Exception("Type '" + type.Name + "' cannot be TrackedAs because it does not derive from Entity or Component");
+            }
+            RefreshTracker(type);
+        }
+
+        public static void RefreshTracker(Type type) {
+            if (typeof(Entity).IsAssignableFrom(type) && !Engine.Scene.Tracker.Entities.ContainsKey(type)) {
+                Engine.Scene.Tracker.Entities.Add(type, new List<Entity>());
```

Comment:
I'd extract `Engine.Scene.Tracker` to a variable OR make this an instance method (applies to all places in this PR that do this)